src/spec-node/dockerCompose.ts

```diff
@@ -107,19 +108,114 @@ export function getRemoteWorkspaceFolder(config: DevContainerFromDockerComposeCo
+	let dockerfile = null;
+	const serviceInfo = getInfoForService(composeService);
+	if (serviceInfo.build) {
+		const { context, dockerfilePath, target } = serviceInfo.build;
+		const resolvedDockerfilePath = cliHost.path.isAbsolute(dockerfilePath) ? dockerfilePath : path.resolve(context, dockerfilePath);
+		dockerfile = (await cliHost.readFile(resolvedDockerfilePath)).toString();
+		if (target) {
+			// Explictly set build target for the dev container build features on that
+			baseName = target;
+		} else {
+			// Use the last stage in the Dockerfile
+			// Find the last line that starts with "FROM" (possibly preceeded by white-space)
+			const { lastStageName, modifiedDockerfile } = ensureDockerfileHasFinalStageName(dockerfile, baseName);
+			baseName = lastStageName;
+			if (modifiedDockerfile) {
+				dockerfile = modifiedDockerfile;
+			}
+		}
+	} else {
+		dockerfile = `FROM ${composeService.image} AS ${baseName}\n`;
+	}
+
+	// determine whether we need to extend with features
+	const labelDetails = async () => { return { definition: undefined, version: undefined }; };
+	const noBuildKitParams = { ...params, buildKitVersion: null }; // skip BuildKit -> can't set additional build contexts with compose
+	const extendImageBuildInfo = await getExtendImageBuildInfo(noBuildKitParams, config, baseName, config.remoteUser ?? 'root', labelDetails);
+
+	let buildOverrideContent = null;
+	if (extendImageBuildInfo) {
+		// Create overridden Dockerfile and generate docker-compose build override content
+		buildOverrideContent = '    build:\n';
+		const { featureBuildInfo } = extendImageBuildInfo;
+		// We add a '# syntax' line at the start, so strip out any existing line
+		const syntaxMatch = dockerfile.match(/^\s*#\s*syntax\s*=.*[\r\n]/g);
+		if (syntaxMatch) {
+			dockerfile = dockerfile.slice(syntaxMatch[0].length);
+		}
+		let finalDockerfileContent = `${featureBuildInfo.dockerfilePrefixContent}${dockerfile}\n${featureBuildInfo.dockerfileContent}`;
+		const finalDockerfilePath = cliHost.path.join(featureBuildInfo?.dstFolder, 'Dockerfile-with-features');
+		await cliHost.writeFile(finalDockerfilePath, Buffer.from(finalDockerfileContent));
+		buildOverrideContent += `      dockerfile: ${finalDockerfilePath}\n`;
+		// remove the target setting as we reference any previous target in the generated override content
+		buildOverrideContent += '      target: \'\'\n';
+
+		if (!serviceInfo.build?.context) {
+			// need to supply a context as we don't have one inherited
+			const emptyDir = cliHost.path.join(await cliHost.tmpdir(), '__devcontainers_cli_empty__');
+			await cliHost.mkdirp(emptyDir);
+			buildOverrideContent += `      context: ${emptyDir}\n`;
+		}
+		// track additional build args to include
+		if (Object.keys(featureBuildInfo.buildArgs).length > 0 || params.buildKitVersion) {
+			buildOverrideContent += '      args:\n';
+			if (params.buildKitVersion) {
+				buildOverrideContent += '        - BUILDKIT_INLINE_CACHE=1\n';
```

Comment:
`BUILDKIT_INLINE_CACHE=1` breaks end users ability to have a docker-compose file with a `build` Dockerfile step and in that `Dockerfile` referencing an image that is not hosted on DockerHub.

This is an end-user breaking change; when people upgrade this extension (I found out by upgrading).

For example, pre-build my image as `my-custom-image` and then the devcontainer `docker-compose.yml` file references a `Dockerfile` which starts with `FROM my-custom-image`.

The error experienced in the devcontainer build process is:

```
failed to solve: my-custom-image: failed to do request: Head "https://registry-1.docker.io/v2/library/my-custom-image/manifests/latest": blah
```

See also https://github.com/moby/buildkit/issues/2343#issuecomment-1867929392

Reply:
@samrocketman  I was reading through the [comment thread](https://github.com/moby/buildkit/issues/2343#issuecomment-1868020509) you linked to - is this still an issue?